packages/agreement-readmodel-writer/test/integration.test.ts

```diff
@@ -0,0 +1,92 @@
+/* eslint-disable functional/no-let */
+/* eslint-disable functional/immutable-data */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { afterEach, beforeAll, describe, expect, it } from "vitest";
+import {
+  AgreementCollection,
+  ReadModelRepository,
+  consumerConfig,
+} from "pagopa-interop-commons";
+import {
+  AgreementAddedV1,
+  AgreementEvent,
+  AgreementEventEnvelope,
+  AgreementStateV1,
+  EventEnvelope,
+} from "pagopa-interop-models";
+import { v4 as uuidv4 } from "uuid";
+import { GenericContainer } from "testcontainers";
+import { handleMessage } from "../src/agreementConsumerService.js";
+
+describe("database test", async () => {
+  let agreements: AgreementCollection;
+
+  const config = consumerConfig();
+  beforeAll(async () => {
+    const mongodbContainer = await new GenericContainer("mongo:4.0.0")
+      .withEnvironment({
+        MONGO_INITDB_DATABASE: config.readModelDbName,
+        MONGO_INITDB_ROOT_USERNAME: config.readModelDbUsername,
+        MONGO_INITDB_ROOT_PASSWORD: config.readModelDbPassword,
+      })
+      .withExposedPorts(27017)
+      .start();
+
+    config.readModelDbPort = mongodbContainer.getMappedPort(27017);
+
+    const readModelRepository = ReadModelRepository.init(config);
+    agreements = readModelRepository.agreements;
+  });
+
+  afterEach(async () => {
+    await agreements.deleteMany({});
+  });
+
+  describe("Handle message for agreement creation", () => {
+    it("should create an agreement", async () => {
+      const id = uuidv4();
```

Comment:
Should we use the new `generateId()` instead of `uuidv4()`?

Reply:
Sure!